src/org/ods/orchestration/util/Project.groovy

```diff
@@ -980,19 +980,22 @@ class Project {
 
     @NonCPS
     DocumentHistory getHistoryForDocument(String document) {
-        return this.data.documentHistories[document]
+        return this.documentHistories[document]
     }
 
     @NonCPS
     DocumentHistory findHistoryForDocumentType(String documentType) {
-        // All docHistories for DTR and TIR should have the same version
-        def key = this.data.documentHistories.keySet().find { it.startsWith(documentType) }
-        return this.getHistoryForDocument(key)
+        def history = getHistoryForDocument(documentType)
+        if (!history) {
+            // All docHistories for DTR and TIR should have the same version
```

Comment:
I find the comment a bit confusing since the implementation of the method nowhere distinguishes between document types. Is this not a general principle?

Reply:
You are right, @metmajer , this method is itself misleading. There is more than one document history for each of these two document types, so it doesn't seem justified to just pick the first one found. The thing is that this method is actually used just to get the document version. As the comment says, the version should be the same in all document histories for the same document type, so it doesn't really matter.
I didn't feel like refactoring further in the scope of a bug fix, but it would be better to property turn this method into one that gets precisely the document version. Would you like me to do that?

Reply:
@jafarre-viewnext yes, please!

Reply:
Done